### ISSUE TITLE
Provide official support and guidance for Docker deployments

### ISSUE THREAD

**Issue by sheluchin** [errbotio/errbot#1293]
### I am...

* [ ] Reporting a bug
* [x] Suggesting a new feature
* [ ] Requesting help with running my bot
* [ ] Requesting help writing plugins
* [ ] Here about something else

### Issue description
There are many different options for deploying and running Errbot, but Docker seems to be among the more discussed strategies. I suggest that we add some documentation about some possible ways that Docker can be used with Errbot, and maybe even include an official Errbot Docker image catered to the most common use cases.

I don't know if it's an official page, but there is an [Errbot Docker Hub][1] page that has not been updated in a number of years. If Docker Hub is going to be part of the strategy, we might also consider using [Automated Builds][3], as it would make keeping the image up to date much easier.

A community member has been maintaining a [docker-errbot][2] repo with the Docker strategy they use for their deployment. It may be helpful reference material, but there are also a few extras there that most users probably do not want or need.

In short, I'm posting this issue to start the discussion. There may be a few challenging parts around Dockerizing an Errbot deployment, such as provisioning and maintaining plugins. It would be nice to get a feel for what the community has tried already and where the pain points might be. Also, we might want to talk about using Kubernetes or some other similar tooling to achieve the same goal of containerized deployment, but Docker seems like the most obvious starting point.

[1]: https://hub.docker.com/r/errbot/err/
[2]: https://github.com/rroemhild/docker-errbot/
[3]: https://docs.docker.com/docker-hub/builds/

**Comment by nitrocode** [comment]
@sijis I was also looking into setting up errbot quickly using a docker container but surprised to see no official one which brought me to this issue. Would love to hear an official status or road map with this on it. Completely understand if more dev volunteer effort is needed.

Thank you.

**Comment by sijis** [comment]
Honestly, I didn't even know there was a docker image published under the org name.

l do think it makes sense to provide, at least, a sample `Dockerfile`. We could potentially use that to create an official image on DockerHub.

I'll look at some Dockerfile i've created for my own and use that as a starting point. I'll probably put up a PR in a couple of days so that I can elicit feedback.

**Comment by sandys** [comment]
hi,
we are requesting for the same. please atleast have an official Dockerfile (that we can build from) and a Docker howto (which should be complete all the way to slack integration for example)

**Comment by maxim-nazarenko** [comment]
hi @sijis ,

I was surprised that there were no official image when I started integration of Errbot into our Docker deployment.
I have created this [image](https://hub.docker.com/repository/docker/mnazarenko/errbot) on docker hub from a personal [github repo](https://github.com/maxim-nazarenko/docker-errbot).

If you have a PR/branch for your Docker support - let me know how I can help you with that.
Or just take pieces from my github repo, that you think are good and suit your needs.

**Comment by nzlosh** [comment]
The most downloaded and longest established docker image for errbot is https://hub.docker.com/r/rroemhild/errbot with >465k docker pulls.  It has been actively maintained for many years (latest update 4months ago) by @rroemhild which seems to be the closest thing to an official image there is.

**Comment by torgeirl** [comment]
> There are many different options for deploying and running Errbot, but Docker seems to be among the more discussed strategies. I suggest that we add some documentation about some possible ways that Docker can be used with Errbot (...)

@sijis: I'm currently deploying my bots as Docker images running as Kubernetes deployments. If you like I could write some generic documentation on making Docker images for running your bots.

**Comment by sijis** [comment]
@torgeirl yeah, if it's something generic absolutely and it can be added to the docs.

I did see a GitHub post about hosting docker images with GitHub actions. Maybe that's something we can do too and all this being tied into the same source.

**Comment by sijis** [comment]
We have been publishing official docker images in https://hub.docker.com/r/errbotio/errbot/tags.
